UserService/src/main/java/org/energygrid/east/userservice/controller/UserController.java

```diff
@@ -20,57 +24,84 @@
     }
 
     @GetMapping()
-    public ResponseEntity<UserViewModel> getUserByUuidOrUsernameOrEmail(@RequestParam(required = false) String uuid, @RequestParam(required = false) String username, @RequestParam(required = false) String email) {
+    public ResponseEntity<UserViewModel> GetUserByUuidOrUsernameOrEmail(@RequestParam(required = false) UUID uuid, @RequestParam(required = false) String username, @RequestParam(required = false) String email) {
         try {
+            String jwt = request.getHeader("jwt");
+            if(jwt == null || jwt.isEmpty()) {
+                throw new IllegalAccessException();
+            }
+
             UserDTO user = userService.getUserByUuidOrUsernameOrEmail(uuid, username, email);
             if (user == null) {
                 return ResponseEntity.status(404).body(null);
             }
 
             var userViewmodel = modelMapper.map(user, UserViewModel.class);
             return ResponseEntity.ok(userViewmodel);
+        } catch (IllegalAccessException e){
+            return ResponseEntity.status(401).body(null);
         } catch (NullPointerException e) {
             return ResponseEntity.status(404).body(null);
         } catch (Exception e) {
+            logger.log(Level.ALL, e.getMessage());
             return ResponseEntity.status(500).body(null);
         }
     }
 
     @PutMapping()
-    public ResponseEntity<?> editUser(@NotNull @RequestBody User user) {
+    public ResponseEntity EditUser(@RequestBody User user) {
         try {
-            userService.editUser(user);
-            return ResponseEntity.ok().build();
+            String jwt = request.getHeader("jwt");
+            if(jwt == null || jwt.isEmpty()) {
+                throw new IllegalAccessException();
+            }
+
+            userService.editUser(user, jwt);
+            return ResponseEntity.ok(null);
+        } catch (IllegalAccessException e){
+          return ResponseEntity.status(401).body(null);
         } catch (NullPointerException e) {
             return ResponseEntity.status(404).body(null);
         } catch (Exception e) {
+            logger.log(Level.ALL, e.getMessage());
             return ResponseEntity.status(500).body(null);
         }
     }
 
-    @DeleteMapping("{uuid}")
-    public ResponseEntity<?> deleteUser(@NotNull @PathVariable String uuid) {
+    @DeleteMapping()
+    public ResponseEntity<?> deleteUser() {
         try {
-            userService.deleteUser(uuid);
+            String jwt = request.getHeader("jwt");
+            if(jwt == null || jwt.isEmpty()) {
+                throw new IllegalAccessException();
+            }
+
+            userService.deleteUser(jwt);
             return ResponseEntity.ok().build();
+        } catch (IllegalAccessException e) {
+            return ResponseEntity.status(401).body(null);
         } catch (Exception e) {
+            logger.log(Level.ALL, e.getMessage());
             return ResponseEntity.status(500).body(null);
         }
```

Comment:
```suggestion
            String jwt = request.getHeader("jwt");
            if(jwt == null || jwt.isEmpty()) {
                throw new IllegalAccessException();
            }

            userService.deleteUser(jwt);
            return ResponseEntity.status(HttpStatus.OK().build();
        } catch (IllegalAccessException e) {
            return ResponseEntity.status(HttpStatus.FORBIDDEN().build();
        } catch (Exception e) {
            logger.log(Level.ALL, e.getMessage());
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
        }
```
Fixed return types for removed wildcard type